dialer.go

```diff
@@ -0,0 +1,53 @@
+package srslog
+
+import (
+	"crypto/tls"
+	"net"
+)
+
+func (w writer) getDialer() func() (serverConn, string, error) {
+	dialers := map[string]func() (serverConn, string, error){
+		"":        w.unixDialer,
+		"tcp+tls": w.tlsDialer,
+	}
+	dialer, ok := dialers[w.network]
+	if !ok {
```

Comment:
It's intended to set the "default" value of `dialer`. The way Go's maps work has a little magic to it. (Which is helpful but annoying that it's magical.)

If there isn't anything in the map that has the given key, `ok` will be `false`. In this case if that happens we want to set it to the `basicDialer`.

This way if they pass in `""` or `"tcp+tls"` we use the special dialers, and if they pass in anything else we use the basic UDP/TCP dialer. (This maintains the same behavior it had before with `if`/`else` blocks.)

This is the only "error" that code can produce.

I can see an argument for being stricter with what you can pass in as a value for `network`. But I wanted to maintain the existing behavior, so I didn't codify the valid values for it. What'll happen if you pass in a bogus network value is that it will fail to dial/connect and you'll get a failure to connect.